contracts/AdjudicationFramework/Push/Feeds.sol

```diff
@@ -0,0 +1,138 @@
+        @param prices The prices for the tokens
+     */
+    function provideInput(
+        address[] calldata tokens,
+        uint256[] calldata prices
+    ) external onlyArbitrator {
+        for (uint256 i = 0; i < tokens.length; i++) {
+            uint256 lastEntry = INPUT_SIZE - 1;
+            for (uint j = 0; j < INPUT_SIZE; j++) {
+                if (
+                    arbitratorInputs[tokens[i]][msg.sender][j].timestamp >
+                    arbitratorInputs[tokens[i]][msg.sender][lastEntry].timestamp
+                ) {
+                    lastEntry = j;
+                } else {
+                    // we can break early, since entries are ordered by timestamp % INPUT_SIZE
+                    break;
+                }
+            }
+            arbitratorInputs[tokens[i]][msg.sender][
+                (lastEntry + 1) % INPUT_SIZE
+            ] = Input(prices[i], block.timestamp);
+        }
+    }
+
+    /** 
+    @dev Provides the latest price
+    @return the latest price
+     */
+    function getPrice(address token) external view returns (uint256) {
+        return getPriceConsideringDelay(token, 0);
+    }
+
+    /** 
+    @dev Provides the latest price considering a delay, that allows other to escalate and freeze wrong oracle inputs
+     */
+    function getPriceConsideringDelay(
+        address token,
+        uint256 deplay
+    ) public view returns (uint256) {
+        address[] memory arbitrators = getAllListMembers();
+        uint256[] memory prices = new uint256[](arbitrators.length);
+        for (uint i = 0; i < arbitrators.length; i++) {
+            if (countArbitratorFreezePropositions[arbitrators[i]] > 0) {
+                continue;
+            }
+            uint256 lastEntry = 0;
+            for (uint j = 0; j < INPUT_SIZE; j++) {
+                if (
+                    arbitratorInputs[token][arbitrators[i]][j].timestamp <
+                    arbitratorInputs[token][arbitrators[i]][lastEntry]
+                        .timestamp &&
+                    arbitratorInputs[token][arbitrators[i]][j].timestamp >
+                    block.timestamp - deplay
+                ) {
+                    lastEntry = j;
+                } else {
+                    // we can break early, since entries are ordered by timestamp % INPUT_SIZE
+                    break;
+                }
+            }
+            prices[i] = arbitratorInputs[token][arbitrators[i]][lastEntry]
+                .price;
+        }
+
+        return calculateAvgerage(prices);
+    }
+
+    /**
+    @dev Calculates the average of a list of prices
+    @param prices The prices used to calculate the average
+     */
+    function calculateAvgerage(
+        uint256[] memory prices
+    ) internal pure returns (uint256) {
+        uint256 sum = 0;
+        uint256 count = 0;
+        for (uint i = 0; i < prices.length; i++) {
+            if (prices[i] != 0) {
```

Comment:
Usually the prices are huge numbers, e.g. with 18 or 6 digits. So we can not represent 0, but easily 0.00001. And then we should set the price nearly to zero. (It will only be a rounding error)